### PR TITLE
Add map renaming to save upgrade logic

### PR DESCRIPTION
**Pull request by kerizane**

### CODE REVIEW

**Comment by kerizane** [comment]
If people want to rename a map they need to add it to MAP_RENAMES.

**Comment by kerizane** [comment]
Or we could just default to the starting location

File: tuxemon/core/save_upgrader.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+"""
+This module is for handling breaking changes to the save file.
+
+Renaming maps:
+  - Increment the value of SAVE_VERSION (e.g. from 1 to 2)
+  - Add an entry to MAP_RENAMES consisting of:
```
Comment (kerizane) [inline tuxemon/core/save_upgrader.py]: Went into lots of detail here as map editors aren't expected to know python.

Comment (bitcraft) [reply tuxemon/core/save_upgrader.py]: good idea

File: tuxemon/core/save_upgrader.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        if new_name:
+            save_data['current_map'] = new_name
+
+
+def upgrade_save(save_data):
+    version = save_data.get("version", 0)
+    for i in range(version, SAVE_VERSION):
+        update_current_map(i, save_data)
+        if i == 0:
+            def fix_items(data):
```
Comment (bitcraft) [inline tuxemon/core/save_upgrader.py]: Is it possible to move this to module scope?

Comment (kerizane) [reply tuxemon/core/save_upgrader.py]: fix_items?

I don't think it's that reusable, but sure.

Comment (bitcraft) [reply tuxemon/core/save_upgrader.py]: its more about readability.  i do like nested functions but it was buried inside a conditional and when i wanted to see where `fix_items` was defined, it wasn't immediately obvious where it was.  putting nested funcs directly after the docstrings is fine, imo.

**Comment by bitcraft** [comment]
This looks like a good start, thanks!  I like the change to its own module.

**Pull request merged**